### ISSUE TITLE
bug: scattered installation instructions for deploying Kubestellar core into a cluster

### ISSUE THREAD

**Issue by KPRoche** [kubestellar/kubestellar#1023]
### Describe the bug

In working on issue #824, I am trying to get Kubestellar core running in a cluster as per
https://docs.kubestellar.io/release-0.7/Coding%20Milestones/PoC2023q1/commands/#deployment-into-a-kubernetes-cluster

That says: 

> Deployment is done with a "kubectl plugin" that creates a [Helm "release"](https://helm.sh/docs/intro/using_helm/#three-big-concepts) in the hosting cluster. As such, it relies on explicit (on the command line) or implicit (in environment variables and/or ~/.kube/config configuration needed to execute Helm commands). An invocation of this plugin starts with kubectl kubestellar deploy and continues with the following, in any order.

The example command below that paragraph is 
`kubectl kubestellar deploy --external-endpoint my-long-application-name.my-region.some.cloud.com:1234` 
and starts with **kubectl kubestellar** which implies that the kubestellar command extensions kubectl have been installed locally. I infer that *that* is the "kubectl plugin" referred to in the quote above.
I'm trying to figure out what has to be installed directly onto my machine, and what gets installed in the created cluster.

### Steps To Reproduce

1. go to https://docs.kubestellar.io/release-0.7/Coding%20Milestones/PoC2023q1/commands/#deployment-into-a-kubernetes-cluster
2. copy and paste the example command into a terminal on your system
3. it will fail because "kubestellar" is not a recognized kubectl command.

### Expected Behavior

A "how to install the kubestellar kubectl plugin" reference link would be really helpful. Ultimately, a page with a from-scratch list of steps to get to the kubestellar-core-in-a-cluster result would be even better.

**Comment by eaepstein** [comment]
While this issue is certainly valid, it is just one of many things deficient in the current documentation. Below is a view of what documentation goals should be. Perhaps this issue could be expanded to address bigger needs.

what do I (as user) need to know to run my workloads via KS
-----------------------------------------------------------
1. To have a simple view of KS architecture
2. How to install KS on a k8s cluster and connect external execution clusters
	a. install prereq and KS binaries
	b. deploy KS on a kind cluster and on other preexisting k8s cluster
	c. configure kubectl to easily access ks-core cluster, ks spaces and external clusters
	d. connect external clusters to KS
3. How to deploy workloads on external clusters
	a. create KS workspaces to create and interact with workloads
	b. customize downsync and/or upsync as needed
	c. deploy a workload to an external cluster
	d. deploy a workload to N clusters using a static yaml specification
4. How to monitor and debug workloads
	a. access workload status
	b. access additional objects created by workload
	c. debug when something does not deploy
	d. debug when status or summarization info is wrong

**Comment by KPRoche** [comment]
OK, an update on basic installation flow process which I am figuring out

- install prerequisites as listed in quickstart page
- install release 0.11 of kcp for your language/architecture and add to $PATH 
    - download from https://github.com/kcp-dev/kcp/releases/tag/v0.11.0
    - expand files
    - add directories to $PATH
- install stable release of kubestellar
    - download from https://github.com/kubestellar/kubestellar/releases
    - expand file
    - add directory to path
- now you can follow instructions for running bare or using kubectl kubestellar deploy into a cluster
    - **NOTE:** if running bare, starting kcp will create the .kcp folder holding admin.kubeconfig as a subfolder of whatever folder you are in when you issue the command
    - in theory, using **kubectl kubestellar deploy** command (with necessary flags and options) will deploy all the core components to either a kind cluster or (if flag is set) an openshift cluster.
    (_I have not succeeded in getting that deploy to work yet_)

**Comment by KPRoche** [comment]
followup: the kubestellar deploy was failing because of a bad kubeconfig context. Mike helped me reset it to a better one, and I succeeded in using the command -- but it wouldn't respond properly to the commands to get its kubeconfig files, nor to a delete command (empty kubestellar namespace stuck forever in "terminating" state)

This pi has microshift installed, I'm going to try creating the cluster with  the openshift flag.

**Comment by clubanderson** [comment]
@KPRoche please add this to the troubleshooting PR you are working on with @namasl

**Comment by KPRoche** [comment]
Regarding item 1, this is my (slightly simplified) version of the architecture:
![KubeStellar structure new v2](https://github.com/kubestellar/kubestellar/assets/25445603/9941503e-a91d-45f9-877f-e2d9d028e9a5)

The Kubestellar core (kubestellar-server) does a join of a workload description with a location description to create a customized workload manifest, which it then uses one of several standard platform controllers to actually deploy along with a corresponding mailbox space. That works with the syncer on the Workload Execution Cluster to complete deployment of the workload there, and to collect/update status/config information from the workload on the WEC. [status/config is a separate data stream from whatever the workload itself is doing]

**Comment by MikeSpreitzer** [comment]
The workload execution cluster has to be a Kubernetes _cluster_.

**Comment by MikeSpreitzer** [comment]
@KPRoche: your remarks above make me wonder whether you found https://docs.kubestellar.io/main/Coding%20Milestones/PoC2023q1/example1/#deploy-kcp-and-kubestellar .  I think it covers pre-reqs, like the quickstart does, and says what you said in https://github.com/kubestellar/kubestellar/issues/1023#issuecomment-1731845729 .

**Comment by KPRoche** [comment]
> The workload execution cluster has to be a Kubernetes _cluster_.

Yes, but was not the final upshot of that conversation there was no apparent reason it could not be the _same_ Kubernetes cluster as the one the Kubestellar core was running on? Ergo resulting in the workload running in a pod in the same cluster.

**Comment by MikeSpreitzer** [comment]
@KPRoche : just to be clear, the reason I am asking about what you found is that I want to get a clear understanding of how the current documentation fails.  It sounds like part of the answer is that some of the problem(s) have already been corrected.  I want to understand what is still problematic.

**Comment by KPRoche** [comment]
> > ... workload running in a pod ...
> 
> Yes, that scans. What alarmed me was earlier when you wrote of the WEC being in a Pod.

Ah -- that is a very good point, and I will make an edit.

**Comment by KPRoche** [comment]
> @KPRoche : just to be clear, the reason I am asking about what you found is that I want to get a clear understanding of how the current documentation fails. It sounds like part of the answer is that some of the problem(s) have already been corrected. I want to understand what is still problematic.

Yes, some of them are definitely much improved already. I am trying to figure out which of my practical problems now are the result of errors on my part vs issues with my host system (rPi) configuration vs issues with documentation.

One of the recent ones I just figured out goes back to the kubectl-plugin behavior... it appears any syntax error for the plugin just returns a "xxx command not found" response from kubectl

**Comment by MikeSpreitzer** [comment]
> it appears any syntax error for the plugin just returns a "xxx command not found" response from kubectl

Yes and no.  The general idea is a tree of command, sub-command, sub-sub-command, and so on.  Any branch of that tree might be implemented by a single executable that itself covers a sub-tree.  If you manage to give enough of a prefix to navigate into such a branch, the error reporting is whatever that branch's executable does.  Otherwise it is what kubectl does.

**Comment by KPRoche** [comment]
This issue is pretty much covered by the new "user quickstart" and I'm going to close it.
